contracts/PartyBid.sol

```diff
@@ -183,7 +198,37 @@ contract PartyBid is Party {
             _ethFee = _closeSuccessfulParty(highestBid);
         }
         // set the contract status & emit result
-        emit Finalized(partyStatus, totalSpent, _ethFee, totalContributedToParty);
+        emit Finalized(partyStatus, totalSpent, _ethFee, totalContributedToParty, false);
+    }
+
+    // ======== External: Expire =========
+
+    /**
+     * @notice Expires an auction, moving it to LOST state and ending the ability to contribute.
+     * @dev Emits a Finalized event upon success; callable by anyone once the expiration date passes.
+     */
+    function expire() external nonReentrant {
```

Comment:
nit but can we add a helper `canExpire` function that returns true/false so it's easy for the front-end to know whether to show an expire button?   this fn could use that

also bonus points for an enum approach so we could easily show why a party can or can not be expired on the front end, a la  https://github.com/PartyDAO/party-dropper/pull/3

Reply:
Cool added that, seems pretty clean

Reply:
Returning the enum and the message in a tuple was the cleanest way to do this internally, and i figure the front-end can just grab the enum and ignore the message since it comes first, lemme know if that feels unergonomic though

Reply:
I would imagine the front-end would rely entirely on the enum value 👍